lib/ex_check/config/default.ex

```diff
@@ -11,6 +11,7 @@ defmodule ExCheck.Config.Default do
     {:ex_doc, "mix docs", detect: [{:package, :ex_doc}]},
     {:ex_unit, "mix test", detect: [{:file, "test"}]},
     {:dialyzer, "mix dialyzer", detect: [{:package, :dialyxir}]},
+    {:unused_deps, "mix deps.unlock --check-unused", detect: [{:elixir, ">= 1.10.0"}]},
```

Comment:
There's some work left to be done here.

1. I've merged #19 to master so please pull it here and ensure the build is green again.

2. We should test this tool in `ExCheck.ProjectCases.BaseTest`. It's version dependent and ex_check runs on Elixir 1.7 and 1.10 on the CI, but that's actually a good thing. I'd add an `if` in test to either check that `unused_deps` was skipped or passed. Although we could go further and test if it fails when there's an unused dep, I would refrain from that since MIx tests should have that covered - for us it's enough to see that the tool gets executed.

3. Documentation for `Mix.Tasks.Check` includes an entire list of tools so we should add something like:

```
  - [`:unused_deps`] - ensures that there are no unused dependencies in the project's `mix.lock`
    file (e.g. after removing a previously used dependency)
```

and a matching documentation link at the end:

```
  [`:unused_deps`]: https://hexdocs.pm/mix/Mix.Tasks.Deps.Unlock.html
```

I'd put it somewhere before the 3rd party tools, e.g. after `formatter`, to keep the mix-based checks together.

4. An entry in `CHANGELOG.md` to highlight the change.

```
## v0.13.0-dev

- **Added** `:unused_deps` tool
```

Reply:
All sounds reasonable.